dotnet-desktop-guide/net/winforms/wfdev-diagnostics/wfdev006.md

```diff
@@ -0,0 +1,65 @@
+---
+title: WFDEV006 warning
+description: Learn about Windows Forms compile-time warning WFDEV006. ContextMenu, DataGrid, MainMenu, Menu, StatusBar, ToolBar are obsolete. They're provided for binary compatibility with .NET Framework.
+ms.date: 03/04/2025
+f1_keywords:
+  - "WFDEV006"
+helpviewer_keywords:
+  - "WFDEV006"
+---
+# Compiler Warning WFDEV006
+
+**Version introduced:** .NET 10 Preview 1
+
+> `ContextMenu`, `DataGrid`, `MainMenu`, `Menu`, `StatusBar`, `ToolBar` are obsolete. They're provided for binary compatibility with .NET Framework.
+
+Starting with .NET 10, some controls are provided for binary compatibility with .NET Framework, but they're marked as obsolete and aren't intended to be used directly from your code. Referencing one of the following controls generates warning `WFDEV006` at compile time:
+
+- <xref:System.Windows.Forms.ContextMenu>
+- <xref:System.Windows.Forms.DataGrid>
+- <xref:System.Windows.Forms.MainMenu>
+- <xref:System.Windows.Forms.Menu>
+- <xref:System.Windows.Forms.StatusBar>
+- <xref:System.Windows.Forms.ToolBar>
+
+Starting with .NET 10, these controls are provided for binary compatibility with .NET Framework, but they're marked as obsolete and aren't intended to be used directly from your code. Use a replacement control instead. Referencing one of these controls warning `WFDEV006` at compile time.
+
+## Workaround
+
+Replace references to these controls with their .NET counterparts:
```

Comment:
```suggestion
Replace references to these controls with their newer counterparts:
```

i.e. the replacements were introduced in .NET Framework2.0, when we documented the "old" controls as obsolete but didn't add compiler warnings.